src/profiler.js

```diff
@@ -0,0 +1,61 @@
+// profiler.js
+
+var _ = require('underscore');
+
+/**
+ * This RDF Pipeline component profiler gathers metrics on the processing time for each
+ * component and records them in the specific metrics directory.  The sequence of execution
+ * is also logged if createLog is called to enable logging. 
+ *
+ * @param node the RDF pipeline component facade
+ */
+module.exports = function(node) { 
+
+    _.extend(node, 
+        {
+          profiler: {
+              metrics: {
+                  averageUpdateTime: 0,
+                  numberOfErrors: 0,
+                  numberOfEvents: 0,
+                  numberOfUpdates: 0,
+                  startTime: Date.now(),
+                  totalErrorTime: 0,
+                  totalProcessingTime: 0,
+                  totalUpdateTime: 0
+              },
+
+              eventTypes: {
```

Comment:
I think the code would be easier to parse if each event type had its own function/entry point. In the same way that loggers have console.error/console.warn it indicate the different log levels, the profiler could have profiler.success(started), profiler.error(started), profiler.notReady(started) to indicate the event type.